File: food-donation-frontend/src/services/deliveryService.js
import apiClient from './apiClient';

const deliveryService = {
  async getDelivery(id) {
    const { data } = await apiClient.get(`/delivery/${id}`);
    return data.data;
  },
  async updateStatus(id, status, notes) {
    const { data } = await apiClient.put(`/delivery/${id}/status`, { status, notes });
    return data.data;
  },
  async confirmPickup(id, otp) {
    const { data } = await apiClient.post(`/delivery/${id}/confirm-pickup`, { otp });
    return data.data;
  },
  async confirmDelivery(id, otp) {
    const { data } = await apiClient.post(`/delivery/${id}/confirm-delivery`, { otp });
    return data.data;
  },
  async getMyDeliveries(params = {}) {
    const { data } = await apiClient.get('/delivery/my-deliveries', { params });
    return data.data;
  },
};

export default deliveryService;

File: food-donation-backend/src/main/java/com/fooddonation/service/DeliveryService.java
package com.fooddonation.service;

import org.springframework.data.domain.Pageable;

import com.fooddonation.dto.response.DeliveryResponse;
import com.fooddonation.dto.response.PageResponse;
import com.fooddonation.enums.AssignmentStatus;

public interface DeliveryService {

	DeliveryResponse assignAgent(Long claimId, Long agentId, String adminEmail);

	DeliveryResponse updateStatus(Long assignmentId, String agentEmail, AssignmentStatus status, String notes);

	DeliveryResponse confirmPickup(Long assignmentId, String agentEmail, String otp);

	DeliveryResponse confirmDelivery(Long assignmentId, String agentEmail, String otp);

	PageResponse<DeliveryResponse> getMyDeliveries(String agentEmail, AssignmentStatus status, Pageable pageable);

	PageResponse<DeliveryResponse> getAllDeliveries(AssignmentStatus status, Pageable pageable);

	DeliveryResponse getById(Long id);

}
